File: couple-app-frontend/src/types/voice_message.ts
export interface VoiceMessage {
  id: number;
  couple_id: number;
  sender_id: number;
  receiver_id: number;
  audio_path: string;
  duration_seconds: number;
  listened_at: string | null;
  created_at: string;
}

export interface VoiceMessageCreate {
  receiver_id: number;
  audio_file: File;
}

export interface VoiceMessageUpdate {
  listened_at: string;
}

File: couple-app/app/schemas/voice_message.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class VoiceMessageBase(BaseModel):
    audio_path: str
    duration_seconds: int
    


class VoiceMessageCreate(VoiceMessageBase):
    receiver_id: int


class VoiceMessageUpdate(BaseModel):
    listened_at: Optional[datetime] = None


class VoiceMessageResponse(VoiceMessageBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    couple_id: int
    sender_id: int
    receiver_id: int
    listened_at: Optional[datetime]
    created_at: datetime
